batch_job/src/batch_job_lambda.c

```diff
@@ -0,0 +1,395 @@
+#include "batch_job_internal.h"
+#include "process.h"
+#include "batch_job.h"
+#include "stringtools.h"
+#include "debug.h"
+#include "jx_print.h"
+#include "jx_parse.h"
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+/*
+These determine whether the down/uploads will all be forked at once
+(in parallel) or one after another (in series). In my tests series
+always performs the best.
+*/
+#define UPLOAD_IN_PARALLEL 0
+#define DOWNLOAD_IN_PARALLEL 0
+
+/*
+This is the file where the Lambda function stores it's
+response. For now there's no good use for it, but it may be
+valuable in the future
+*/
+static const char *output_file_name = "/dev/null";
+
+/*
+Decides what to name the folder where files will be uploaded to and
+downloaded from for this particular job.
+*/
+char* bucket_folder_create(int pid) {
+  char* bucket_folder = malloc(sizeof(*bucket_folder) * 256);
+  sprintf(bucket_folder, "%d", pid);
+  return bucket_folder;
+}
+
+/*
+Forks an upload process and waits, or forks them all at once,
+depending on UPLOAD_IN_PARALLEL
+*/
+int upload_file(const char *file_name, const char *profile_name, const char *region_name, const char* bucket_name, const char *bucket_folder) {
+  int jobid = fork();
+  /* parent */
+  if (jobid > 0)  {
+    int status;
+    if (UPLOAD_IN_PARALLEL) {
+      return jobid;
+    }
+    else {
+      waitpid(jobid,&status,0);
```

Comment:
Careful here: always check the result of waitpid.  You may need to retry if result=1 and errno=EINTR